packages/iTwinUI-react/src/core/ComboBox/ComboBoxInput.tsx

```diff
@@ -128,28 +136,41 @@ export const ComboBoxInput = React.forwardRef(
               prevIndex = Number(prevElement?.getAttribute('data-iui-index'));
 
               if (prevElement?.ariaDisabled !== 'true') {
-                return dispatch(['focus', prevIndex]);
+                return dispatch({ type: 'focus', value: prevIndex });
               }
             } while (prevIndex !== focusedIndexRef.current);
             break;
           }
           case 'Enter': {
             event.preventDefault();
             if (isOpen) {
-              dispatch(['select', focusedIndexRef.current]);
-              dispatch(['close']);
+              if (multiple) {
+                if (focusedIndexRef.current > -1) {
+                  dispatch({
+                    type: 'multiselect',
+                    value: focusedIndexRef.current,
+                  });
+                  onChangeHandler && onChangeHandler(focusedIndexRef.current);
```

Comment:
nit: why not use `?.`?

i.e. `onChangeHandler?.(focusedIndexRef.current)`

same thing below in line 159

Reply:
Added this magic.